scrap_emails.py

```diff
@@ -0,0 +1,125 @@
+
+
+    def __init__(self, site='', verify_endpoint=True, *args, **kwargs):
+        """Initializer for the spider that sets up the sites to parse. First uses Request to get the correct base url
+        for a passed site. Then sets any other passed parameters this may choose to implement.
+
+        Args:
+            site: A site to get emails from. In the future, this could be made into a list.
+
+            verify_endpoint: Whether to verify that a redirected link is still under the initial url domain..
+        """
+
+        if not site.startswith('http'):
+            site = 'http://' + site
+
+        r = requests.get(site)
+        if r.status_code == 200:
+            self.start_urls.append(urlparse.urljoin(r.url, '/')[0:-1])
+            self.existing_visited.append(urlparse.urljoin(r.url, '/')[0:-1])
+        else:
+            raise requests.RequestException('Site %s could not be resolved to a HTTP 200 status code' % site)
+
+        # FIXME: Doesn't seem like one can pass a boolean from the command line. Could do like --no-verify as an argument
+        # but easier for now to keep this syntax.
+        if verify_endpoint == "False":
+            self.verify_redirect_endpoint = False
+        else:
+            self.verify_redirect_endpoint = True
+
+        super(EmailSpider, self).__init__(*args, **kwargs)
+
+
+    def parse(self, response):
+        """The main parsing logic of this crawler.
+
+        Args:
+            response: The response from resolving a remote url.
+
+        Returns:
+            A hash of {email: email_value} that is aggregated into an end output.
+        """
+
+        if self.verify_redirect_endpoint and not response.url.startswith(tuple(self.start_urls)):
+            return
+
+        for email in self.get_emails(response.text):
+            yield {'email': email}
+        for link in response.css('a::attr(href)').extract():
+            link = response.urljoin(link)
+
+            #yield {'link': link, 'start_url': self.start_urls[0]}
+
+            # First check as the setting doesn't seem to work as expected in scrapy. Will keep trying to load links after
+            # MAX_DEPTH even if it isn't actually allowing them to process further.
+            if response.meta["depth"] < self.custom_settings['DEPTH_LIMIT'] and self.is_valid_link(link) is True:
+
+                # Add this url to the visited list. Exclude any trailing slash marks for consistency.
+                self.existing_visited.append(re.sub(r'\/$', '', link))
+                yield scrapy.Request(link, callback=self.parse,)
+
+
+    def is_valid_link(self, l):
+        guessed_type = mimetypes.guess_type(l.split('?')[0])
+        # Ensure that we don't parse things like images or pdfs
+        if guessed_type[0] is None or guessed_type[0].startswith('text'):
+            # Ensure that we stick only to this domain.
+            # FIXME: There is an issue with http vs https in this sample application. Also with www vs non-www.
```

Comment:
Instead of comparing against the full base url, I've stripped out the "http(s)://" portion of it. Visited urls also only store the http-less portion of their url now which has the added benefit of saving on memory space.